backend/src/utils/route-security.ts

```diff
@@ -0,0 +1,114 @@
+import { getDashboardConfig } from './resourceUtils';
+import { K8sResourceCommon, KubeFastifyInstance, OauthFastifyRequest } from '../types';
+import { getUserName } from './userUtils';
+import { createCustomError } from './requestUtils';
+import { FastifyReply, FastifyRequest } from 'fastify';
+
+const usernameTranslate = (username: string): string => {
+  const encodedUsername = encodeURIComponent(username);
+  return encodedUsername
+    .replace(/!/g, '%21')
+    .replace(/'/g, '%27')
+    .replace(/\(/g, '%28')
+    .replace(/\)/g, '%29')
+    .replace(/\*/g, '%2a')
+    .replace(/-/g, '%2d')
+    .replace(/\./g, '%2e')
+    .replace(/_/g, '%5f')
+    .replace(/~/g, '%7f')
+    .replace(/%/g, '-')
+    .toLowerCase();
+};
+
+const getNamespaces = (
+  fastify: KubeFastifyInstance,
+): { dashboardNamespace: string; notebookNamespace: string } => {
+  const config = getDashboardConfig();
+  const notebookNamespace = config.spec.notebookController?.notebookNamespace;
+  const fallbackNamespace = config.metadata.namespace || fastify.kube.namespace;
+
+  return {
+    notebookNamespace: notebookNamespace || fallbackNamespace,
+    dashboardNamespace: fallbackNamespace,
+  };
+};
+
+const requestSecurityGuard = async (
+  fastify: KubeFastifyInstance,
+  request: OauthFastifyRequest,
+  name: string,
+  namespace: string,
+): Promise<void> => {
+  const { notebookNamespace, dashboardNamespace } = getNamespaces(fastify);
+  const username = await getUserName(fastify, request);
+  const translatedUsername = usernameTranslate(username);
+
+  if (dashboardNamespace === notebookNamespace || namespace === dashboardNamespace) {
+    // Requesting something from the exclusive dashboard namespace -- can't validate cleanly
+    return;
+  }
+
+  if (namespace !== notebookNamespace) {
```

Comment:
minor, but I find these guards hard to read.
Could be something like:
```
  if (namespace === dashboardNamespace) {
    // Requesting something from the dashboard namespace -- can't validate cleanly
    return;
  }

  if (namespace !== notebookNamespace) {
    ...
    throw createCustomError(
      'Wrong namespace',
      'Cannot request a resource outside of the notebook namespace',
      403,
    );
  }

  if (!name.includes(translatedUsername)) {
    ...
  }
  
return // namespace and name are valid, authorize the request # have this explicit here
```

Reply:
I don't follow your request -- is the only thing you added a comment at the bottom with an empty return?

Reply:
yeah that's minor, maybe just bikeshedding, but I find it hard to read these two tests:
```
  if (dashboardNamespace === notebookNamespace || namespace === dashboardNamespace) {...}
  if (namespace !== notebookNamespace) {...}
```
and I don't really understand which case is supposed to reach the end of the function